hiredis.c

```diff
@@ -563,15 +563,15 @@ void redisFreeCommand(char *cmd) {
     free(cmd);
 }
 
-void __redisSetError(redisContext *c, int type, const char *str) {
-    size_t len;
+void __redisSetError(redisContext *c, int type, const char *format, ...) {
+    va_list args;
 
     c->err = type;
-    if (str != NULL) {
-        len = strlen(str);
-        len = len < (sizeof(c->errstr)-1) ? len : (sizeof(c->errstr)-1);
-        memcpy(c->errstr,str,len);
-        c->errstr[len] = '\0';
+    if (format != NULL) {
+        va_start(args, format);
+        /* We ignore return value since SetError has no return code */
+        (void) vsnprintf(c->errstr, sizeof(c->errstr), format, args);
```

Comment:
I am unsure if we are still aiming for C89 compliance. If we are, then `vsnprintf` is verbotten, since it is C99. Maybe @michael-grunder can tell us more.

Reply:
Reverted this change since I don't use it in this commit anyways

Reply:
It's no longer important vis-a-vis this commit but AFAIK hiredis has always required C99, so it's safe to do things that won't work in C89/C90.

Reply:
Looking at the code, I see classic marks of C89, for example, no in-line declarations of variables, and no usages of designated initializers. If hiredis was ever intended to be compiled with VS 2008, then it's C89

Reply:
You could totally be right that hiredis once targeted C89, but if it did it was before I became aware of the project.

It seems to be C99 as far back as [4/4/2011](https://github.com/redis/hiredis/issues/31#issuecomment-952785)

I have no strong feelings on the matter but if we decided to support C89 we'd have to start maintaining two copies of things (e.g. sds strings).

Reply:
good to know!